routes.js

```diff
@@ -1,58 +1,79 @@
+const fs = require('fs');
 
-// Middleware
-router.use(express.json());
+// Fetch data from the members.json file
+const loadMembers = () => {
+    const jsonData = fs.readFileSync(__dirname + "/" + "members.json");
+        return JSON.parse(jsonData);
+}
 
-// Get method (Get all member)
-router.get('/getAllMembers', (req, res) => {
-  res.send(teamMembersArray);
-  if(!error) {
-      res.status(200);
-  }
-});
+// Update members.js file
+const saveMembers = (newData) => {
+    const stringifyData = JSON.stringify(newData);
+    fs.writeFileSync(__dirname + "/" + "members.json", stringifyData)
+}
 
-// Post method (Add a new member)
-router.post('/addNewMember', (req, res) => {
-  const member = {
-    memberId: req.body.memberId,
-    role: req.body.role,
-    firstname: req.body.firstname,
-    lastname: req.body.lastname
-  }
-  teamMembersArray.push(member);
-  res.send(member);
-  if(!error) {
-    res.status(200);
-  }
-});
+// Endpoint to get a list of all team members (get request)
+createMyCustomApi.get('/getAllMembers', function(req, res) {
+    fs.readFile(__dirname + "/" + "members.json", 'utf8', function(err, data) {
+        res.status(200);
+        res.end(data);
+    });
+})
 
-// Put method (Update an existing member)
-router.put('/updateExistMember/:memberId', (req, res) => {
-  res.send(`Update member ${memberId}`);
+// Endpoint to add a new member to the team (post request)
+createMyCustomApi.post('/addNewMember', (req, res) => {
+    const newMember = {
+        memberId: req.body.memberId,
+        role: req.body.role,
+        firstname: req.body.firstname,
+        lastname: req.body.lastname
+    }
+    const members = loadMembers();
+    if(members.some(member => member.memberId === newMember.memberId)) {
+        res.status(409).json({ message: 'This record ID already exists...' });
+    }
+    else {
+        members.push(newMember);
+        saveMembers(members);
+        res.status(200).json({ message: 'OK' });
+    }
 });
 
-// Delete method (Delete member from the records)
-router.delete('/removeMember/:memberId', (req, res) => {
-  const memberId = req.params.id;
-  res.send(`Delete member ${memberId}`);
+// Endpoint to update the information about an existing team member (put request)
+createMyCustomApi.put('/updateExistMember/:memberId', (req, res) => {
+    const putId = parseInt(req.params.memberId, 10);
+    let members = loadMembers();
+    if(members.some(member => member.memberId === putId)) {
+        const filteredData = members.filter((i) => i.memberId === putId);
+        filteredData.forEach((i) => {
+            i.memberId = req.body.memberId,
+            i.role = req.body.role,
+            i.firstname = req.body.firstname,
+            i.lastname = req.body.lastname
+        });
+        saveMembers(members);
+        res.status(201).json({ message: 'OK' });
+    }
+    else {
+        res.status(404).json({ message: 'The record with the required ID does not exist...' });
```

Comment:
Could you please invert this if-statement, so the validation goes before the happy flow? Also, once inverted you don't really need to wrap the happy flow into `else` section. since `res.status(404)` will be exiting earlier

Reply:
I did it, but I kept the else section because, without it I receive an error (Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client), although the API is still functional.
May I update the ‘delete’ method as well?

Reply:
Solved during last Coding Q&A. Please don't forget to apply the solution to the rest of endpoints here